core/commit.go

```diff
@@ -154,15 +154,23 @@ func makeCommitmentCounter(f uint32) commitmentCounter {
 		if replicasCommitted[replicaID] {
 			return false, fmt.Errorf("Duplicated commitment")
 		}
-		replicasCommitted[replicaID] = true
 
-		if len(replicasCommitted) == int(f+1) {
-			delete(prepareStates, prepareCV)
-			lastDoneCV++
+		for cv := lastDoneCV + 1; cv < prepareCV; cv++ {
+			s := prepareStates[cv]
+			if len(s) != 0 && !s[replicaID] {
+				return false, fmt.Errorf("Skipped commitment")
+			}
+		}
 
-			return true, nil
+		if len(replicasCommitted) < int(f) {
+			replicasCommitted[replicaID] = true
+
+			return false, nil
 		}
 
-		return false, nil
+		lastDoneCV = prepareCV
+		delete(prepareStates, prepareCV)
```

Comment:
This commit should make it impossible. Because it does not allow to skip commitments, the threshold of `f+1` commitment will be reached without skipping any `prepareCV`.

Reply:
We can also double-check this, and panic in case the check fails.